### DISCUSSION TITLE
Extracting from terribly formatted PDF

### DISCUSSION THREAD

**Discussion by sarahbton** [jsvine/pdfplumber#837] [Ask for help with specific PDFs]
Any advice on extracting the property information in the PDF attached. Preferably each data element listed for every property but the assessment and tax information is not critical. For the first property the output would be "Map Reference" = "0064 17 0024", "Owner Name & Address" = "[NAME IN PDF] 11620 Cedar...", etc. There are four separate property records on the page.

Thank you in advance.

[examplepage.pdf](https://github.com/jsvine/pdfplumber/files/10948758/examplepage.pdf)

**Comment by jsvine** [comment]
Hi @sarahbton, and thanks for your interest in `pdfplumber`. My general strategy with PDFs like those:

- Use information from `page.lines` and/or `page.rects` (depending on the PDF) to identify those solid horizontal lines that separate each property's information
- Use the positional information for those lines in combination with `page.crop(...)` to isolate each property's section of the PDF
- Then, for a PDF like this one (where the text within each section looks to be fairly consistent), use `cropped_page.extract_text(layout=True)` to extract the text for each section
- Use regular expressions to pull out the individual fields that extracted text
